### PR TITLE
More Documentation

### PR DESCRIPTION
**Pull request by zicklag**
This PR follows up on #11 with more documentation, examples, etc.

It's based on #11 so all of the extra commits shown now won't be in it once #11 is merged.

### CODE REVIEW

**Comment by PsichiX** [comment]
hi! are these changes ready for review or are they WIP and gonna be published with next version (not one published today)? :D

**Comment by zicklag** [comment]
In just a second I'm going to push another commit with just a bit more documentation then these these might as well be merged into the next release. Should just take me a few minutes to finish the file I'm on.

**Comment by zicklag** [comment]
There you go!

---

All this documenting does wonders for how much I understand this library. LOL :D

File: raui-core/src/data_binding.rs [hunk 1]
```diff
@@ -86,7 +187,10 @@ where
         }
     }
 
+    /// Set the inner data directly and notify the [`ChangeNotifier`] ( if set )
     pub fn write(&mut self, v: T) {
+        // TODO: This will silently swallow any failure to obtain the RwLock, should we return a
```
Comment (PsichiX) [inline raui-core/src/data_binding.rs:192]: we can discuss implementing result for the next version later, but for this version we can go with silent failure to not block the update - there will be other `write` methods in other types that might need to return Result too so we can make a bulk change for all writes in RAUI.

**Review by PsichiX — APPROVED** [review]
i've left one note about the future changes regarding Result for all `write` methods in RAUI.
really great job! approved 🚀

File: raui-core/src/data_binding.rs (hunk 1 above)
Comment (zicklag) [reply raui-core/src/data_binding.rs:192]: OK, cool, that's what I was thinking. :+1:

**Comment by zicklag** [comment]
> really great job!

Thank you! :smiley:

**Pull request merged**